doc/src/tutorial/basic_operations.rst

```diff
@@ -197,4 +242,239 @@ dictionary of ``sympy_name:numerical_function`` pairs.  For example
     >>> f(0.1)
     0.1
 
+
+Functions
+=========
```

Comment:
I would create a new document on creating custom functions for this (perhaps in special_topics, unless you can think of a better place for it).